blocks/byline-block/features/byline/default.jsx

```diff
@@ -1,125 +1,43 @@
+  if (!props[propName] || (props[propName] && typeof props[propName] === 'string' && allowable.includes(props[propName]))) {
+    return null;
   }
 
-  render() {
-    const { stylesFor } = this.props;
-    const { by, arcSite } = this.state;
-
-    const authors = by.length > 0 && by.map((author) => {
-      if (author.type === 'author') {
-        /* eslint-disable-next-line camelcase */
-        const authorName = author?.additional_properties?.original?.byline || author?.name;
-        const hasURL = Object.prototype.hasOwnProperty.call(author, 'url');
-
-        // If the author has a url to their bio page, return an anchor tag to the bio.
-        // If not, just return the string.
-        if (authorName) {
-          return (hasURL) ? `<a href="${author.url}">${authorName}</a>` : authorName;
-        }
-        // Those without name will not be included in the byline
-      }
-
-      return null;
-    }).filter((author) => author != null);
-
-    const numAuthors = authors.length;
-    // This will be an innerHTML to accommodate potential multiple anchor tags within the section
-    // Leave it empty so that if there's no author with listed name it would just return '' string
-    // note: default is empty string with one space
-    let bylineString = ' ';
-
-    // Depending on how many authors there are, change style accordingly
-    if (numAuthors) {
-      switch (numAuthors) {
-        case 1: {
-          bylineString += `${authors[0]}`;
-          break;
-        }
-        case 2: {
-          bylineString += `${authors[0]} ${this.phrases.t('byline-block.and-text')} ${authors[1]}`;
-          break;
-        }
-        default: {
-        // Iterate through each of the authors until the last two
-          for (let i = 0; i < numAuthors - 2; i += 1) {
-            bylineString += `${authors[i]}, `;
-          }
-
-          // Add last two authors in Oxford comma style
-          bylineString = `${bylineString}${authors[numAuthors - 2]} ${this.phrases.t('byline-block.and-text')} ${authors[numAuthors - 1]}`;
-          break;
-        }
-      }
-    }
-
-    // note: default is empty string with one space
-    // would not want to return 'by' all by itself unless it's by Anonymous ;)
-    if (bylineString === ' ') {
-      return null;
-    }
-
-    return (
-      <BylineSection primaryFont={getThemeStyle(arcSite)['primary-font-family']} className="byline" stylesFor={stylesFor}>
-        <By stylesFor={stylesFor}>{this.phrases.t('byline-block.by-text')}</By>
-        <BylineNames dangerouslySetInnerHTML={{ __html: `${bylineString}` }} stylesFor={stylesFor} />
-      </BylineSection>
-    );
-  }
-}
+  return new Error(`When included, ${propName} (optional) in ${componentName} must be one of the following strings: "${allowable.join('" | "')}"`);
+};
 
 ArticleByline.label = 'Byline – Arc Block';
 
 ArticleByline.propTypes = {
   story: PropTypes.object,
-  stylesFor: PropTypes.string,
+  stylesFor: stylesForPropValidation,
```

Comment:
The `ArticleByline` is an Arc Block so should only have `customField` prop types, as these are the only ones exposed to an editor via PageBuilder.
The `Byline` in `@wpmedia/shared-styles` is the reusable component that should have these prop types.